android-sdk/src/main/java/com/mobilecoin/lib/FogReport.java

```diff
@@ -0,0 +1,114 @@
+
+final class FogReport extends Native {
+    private final static String TAG = FogReport.class.getName();
+    private final long publicKeyExpiry;
+
+    FogReport(@NonNull String fogReportId,
+              @NonNull VerificationReport verificationReport,
+              long publicKeyExpiry
+    ) throws FogReportException {
+        this.publicKeyExpiry = publicKeyExpiry;
+        try {
+            init_with_verification_report(fogReportId, verificationReport, publicKeyExpiry);
+        } catch (Exception exception) {
+            throw new FogReportException("Unable to create report from the provided arguments",
+                    exception);
+        }
+    }
+
+    FogReport(@NonNull String fogReportId,
+              @NonNull MobileCoinAPI.DcapEvidence dcapEvidence,
+              long publicKeyExpiry
+    ) throws FogReportException {
+        this.publicKeyExpiry = publicKeyExpiry;
+        try {
+            init_with_dcap_evidence(fogReportId, dcapEvidence.toByteArray(), publicKeyExpiry);
+        } catch (Exception exception) {
+            throw new FogReportException("Unable to create report from the provided arguments",
+                    exception);
+        }
+    }
+
+    @NonNull
+    static FogReport fromProtoBuf(@NonNull ReportOuterClass.Report proto) throws FogReportException {
+        Logger.i(TAG, "Deserializing report from protobuf");
+        if(!proto.hasDcapEvidence()) {
+            // Initialize from VerificationReport
+            // Signature
+            VerificationSignature verificationSignature =
+                    new VerificationSignature(proto.getVerificationReport().getSig().getContents().toByteArray());
+
+            // Chain list
+            List<ByteString> chainList = proto.getVerificationReport().getChainList();
+            byte[][] chain = new byte[chainList.size()][];
+            for (int i = 0; i < chainList.size(); i++) {
+                chain[i] = chainList.get(i).toByteArray();
+            }
+
+            // VerificationReport
+            VerificationReport verificationReport = new VerificationReport(
+                    verificationSignature,
+                    chain,
+                    proto.getVerificationReport().getHttpBody());
+
+            // Report
+            return new FogReport(
+                    proto.getFogReportId(),
+                    verificationReport,
+                    proto.getPubkeyExpiry()
+            );
+        }
+        else {
+            // Initialize from DcapEvidence
+            return new FogReport(
+                    proto.getFogReportId(),
+                    proto.getDcapEvidence(),
+                    proto.getPubkeyExpiry()
+            );
+        }
+    }
+
+    public long getPublicKeyExpiry() {
+        Logger.i(TAG, "Getting public key expiry", null, publicKeyExpiry);
+        return publicKeyExpiry;
+    }
+
+    @Override
+    protected void finalize() throws Throwable {
+        try {
+            finalize_jni();
```

Comment:
Is this missing a `if (rustObj != 0) {`?

Reply:
Good find! But that case should be handled by the catch block.